src/types/domain.py

```diff
@@ -26,6 +26,12 @@ class UserEnriched(User):
     leader: bool | None = None
     follower: bool | None = None
 
+class UserFirebase(User):
+    firebase_id: str
+
+class UserAuth(User):
```

Comment:
I don't like having so many variants of the User class. I wonder if we should separate the access-token from the User. Instead, make it an Auth object. If needed, we can have endpoints that return both a User and an Auth.

Reply:
I'll leave this one to you.